packages/fx-core/src/core/FxCore.ts

```diff
@@ -1240,51 +1242,130 @@ export class FxCore {
     return await coordinator.publishInDeveloperPortal(context, inputs as InputsWithProjectPath);
   }
 
+  hasActionWithName(provisionNode: any, action: string, name: string): any {
```

Comment:
Yaml file related API should moved into a sub-component in `src/configManager` to avoid code bloating in FxCore.